modules/cloud-build-v2-connection/variables.tf

```diff
@@ -0,0 +1,181 @@
+      host_uri                                  = string
+      read_authorizer_credential_secret_version = string
+      service                                   = optional(string)
+      ssl_ca                                    = optional(string)
+      webhook_secret_secret_version             = optional(string)
+    }))
+    github = optional(object({
+      app_installation_id                  = optional(string)
+      authorizer_credential_secret_version = optional(string)
+    }))
+    github_enterprise = optional(object({
+      app_id                        = optional(string)
+      app_installation_id           = optional(string)
+      app_slug                      = optional(string)
+      host_uri                      = string
+      private_key_secret_version    = optional(string)
+      service                       = optional(string)
+      ssl_ca                        = optional(string)
+      webhook_secret_secret_version = optional(string)
+    }))
+    gitlab = optional(object({
+      host_uri                                  = optional(string)
+      webhook_secret_secret_version             = string
+      read_authorizer_credential_secret_version = string
+      authorizer_credential_secret_version      = string
+      service                                   = optional(string)
+      ssl_ca                                    = optional(string)
+    }))
+  })
+  default  = {}
+  nullable = false
+  validation {
+    condition = (
+      (try(var.connection_config.bitbucket_cloud, null) == null ? 0 : 1) +
+      (try(var.connection_config.bitbucket_data_center, null) == null ? 0 : 1) +
+      (try(var.connection_config.github, null) == null ? 0 : 1) +
+      (try(var.connection_config.github_enterprise, null) == null ? 0 : 1) +
+      (try(var.connection_config.gitlab, null) == null ? 0 : 1) == 1
+    )
+    error_message = "One and only one of bitbucket_cloud, bitbucket_data_center, github, github_enterprise, gitlab can be defined."
+  }
+}
+
+variable "connection_create" {
+  description = "Create connection."
+  type        = bool
+  default     = true
+}
+
+
+variable "context" {
+  description = "Context-specific interpolations."
+  type = object({
+    custom_roles   = optional(map(string), {})
+    iam_principals = optional(map(string), {})
+    locations      = optional(map(string), {})
+    project_ids    = optional(map(string), {})
+  })
+  default  = {}
+  nullable = false
+}
+
+variable "disabled" {
+  description = "Flag indicating whether the connection is disabled or not."
+  type        = bool
+  default     = false
+}
+
+variable "location" {
+  description = "Location."
+  type        = string
+}
+
+variable "name" {
+  description = "Name."
+  type        = string
+}
+
+variable "prefix" {
+  description = "Prefix."
+  type        = string
+  default     = null
+  validation {
+    condition     = var.prefix != ""
+    error_message = "Prefix cannot be empty, please use null instead."
+  }
+}
```

Comment:
Is the requirement that connection names need to be globally unique? Or do they need to be only unique within the project? If the latter is the case, then I think we don't need the `prefix`